plugins/string/plugin.spec.yaml

```
@@ -172,3 +172,31 @@ actions:
         title: Length
         description: Length of string
         required: true
+  replace:
+    title: Replace
+    description: Replace parts of a string
+    input:
+      in_string:
+        title: In String
+        description: The string to replace parts of
+        type: string
+        required: true
+        example: "this is a string"
+      string_part_to_find:
+        title: String Part to Find
+        description: The string part to look for. All instances of this string will be replaced
+        type: string
+        required: true
+        example: "string"
+      replacement_value:
+        title: Replacement Value
+        description: The string that will replace the parts that are found. If left blank the characters to find will be deleted
+        type: string
+        required: false
+        example: "replacement"
+    output:
+      result_string:
+        title: Result String
+        description: The string after replacement
+        type: string
+        required: false
```

```suggestion
        required: true
```
Check to make sure an empty string validates fine in product